cpp/src/parquet/arrow/arrow_reader_writer_test.cc

```diff
@@ -3796,6 +3796,95 @@ TEST(TestArrowWriterAdHoc, SchemaMismatch) {
   ASSERT_RAISES(Invalid, writer->WriteTable(*tbl, 1));
 }
 
+// ----------------------------------------------------------------------
+// Tests for directly writing DictionaryArray
+class TestArrowWriteDictionary : public ::testing::Test {
+ public:
+  void SetUp() override {
+    properties_ = default_arrow_writer_properties();
+    serialized_data_ = AllocateBuffer();
+  }
+
+  // Generates a range of single character strings from start to end (inclusive)
+  // Nulls will be inserted starting at index 0.  If there are any null
+  // values then start will not be the true min.  However, the dictionary
+  // array will always contain all the values even if they aren't all used.
+  void GenerateRange(int num_nulls, char start, char end) {
```

Comment:
this seems overly complex, could you construct the arrays in question with JSON and then combine into a single test case that verifies all of the statistics?

Reply:
Yes, I had grander things in mind and then didn't need them.  I simplified the test considerably and added cases for some of the other binary types.